app/Domain/KeyVisibilityUpgrades/KeyVisibilityUpgrade.php

```diff
@@ -31,4 +31,27 @@ public function getCreatedAt(): ?DateTimeImmutable
     {
         return $this->createdAt;
     }
+
+    private function getDaysLeft(): int
+    {
+        if ($this->createdAt === null) {
+            throw new \Exception('Property createdAt should exist');
+        }
+
+        $days = $this->createdAt->diff(new DateTimeImmutable())->days;
```

Comment:
I think this is the days passed since the creation, not the days until the old key expires.

Reply:
To be clear the date of expiration is not stored, because that would be tricky when the grace period changes. So an easy work around is to just store the creation date of the upgrade request and use the controller or UI to take into account the grace period.